libminifi/src/core/ProcessSession.cpp

```diff
@@ -212,6 +212,10 @@ void ProcessSession::transfer(const std::shared_ptr<core::FlowFile> &flow, Relat
 }
 
 void ProcessSession::write(const std::shared_ptr<core::FlowFile> &flow, OutputStreamCallback *callback) {
+  gsl_Expects(_updatedFlowFiles.contains(flow->getUUID())
+              || _addedFlowFiles.contains(flow->getUUID())
+              || std::any_of(_clonedFlowFiles.begin(), _clonedFlowFiles.end(), [&flow](const auto& cloned_flow_file) { return flow == cloned_flow_file; }));
+
```

Comment:
This check looks like it's on the more expensive side. Consider changing it to [`gsl_ExpectsAudit`](https://github.com/gsl-lite/gsl-lite#contract-checking-configuration-macros) to remove the check from non-CI builds that don't define `gsl_CONFIG_CONTRACT_CHECKING_AUDIT`. (same in append)

Reply:
good idea, changed it in https://github.com/apache/nifi-minifi-cpp/pull/1226/commits/300532be1a9347d2926c139185bbf46edb619102

Reply:
Consider extracting the lambda to a variable to avoid the error with clang. It will start working once they implement the C++20 feature "Lambdas in unevaluated contexts".
This is the disabled implementation of the macro:
```
# define gsl_ELIDE_( x )  static_assert( ::std::is_constructible<bool, decltype( x )>::value, "argument of contract check must be convertible to bool" )
```